Makefile

```diff
@@ -121,3 +129,27 @@ endif
 		  --object-dir=build/$(TEST_BUILD_ID) \
 		  --output-dir=build/update_files/$(TEST_BUILD_ID) \
 	      $@.sdef
+
+# Rule for generating legato.cwe files from .update files.
+CWE_FILES = $(foreach board,$(BOARDS),build/$(board)_$(LEGATO_TARGET)/legato.cwe)
+$(CWE_FILES): build/%_$(LEGATO_TARGET)/legato.cwe: $(UPDATE_FILE_DIR)/%.$(LEGATO_TARGET).update
+	systoimg -s $(LEGATO_TARGET) $< build/yellow_wp76xx
+
+# Goals, like "yellow_spk", for building complete .spk files for factory programming.
+# The names of the resulting .spk files are of the form build/yellow_wp76xx.spk.
+# NOTE: This requires leaf.
+# NOTE: Sadly, there's some inconsistencies in the naming of leaf packages that means the rules
+#       have to be hand-written for each supported board_module combination.
+FACTORY_SPK_GOALS = $(foreach board,$(BOARDS),$(board)_spk)
+.PHONY: $(FACTORY_SPK_GOALS)
+$(FACTORY_SPK_GOALS): %_spk: % build/%_$(LEGATO_TARGET).spk
+
+build/yellow_wp76xx.spk: build/yellow_wp76xx/legato.cwe
+	swicwe -c $(LEAF_DATA)/wp76-linux-image/linux.cwe \
```

Comment:
This is untested, but what do you think of this?
```
swicwe -c \
    $(WP76XX_LINUX_IMAGES)/linux.cwe \
    $(WP76XX_MODEM_IMAGES)/*_GENERIC_*.spk
    $< \
    -o $@
```

It's not so hard-coded to one specific release.

Reply:
Yeah, that's probably better.

Reply:
I'm leaving the part about creating variables like WP76XX_MODEM_IMAGES to a later date. I figure we can add that complexity when we start adding support for more modules. (WP77 soon to come.)

Reply:
ok, we can clean it up later